### PR TITLE
add unity config

### PR DESCRIPTION
**Pull request by SimonCropp**
#skip-changelog

so we can have custom config / apis for unity

### CODE REVIEW

**Comment by SimonCropp** [comment]
@bruno-garcia if this looks good to u, can u merge and update the module in unity

**Comment by bruno-garcia** [comment]
I believe we can define properties in the unity directory.build.props in the unity repo since we have the submodule as a sub folder. Would that be a better approach?

**Comment by bruno-garcia** [comment]
On a second thought it won't because it already includes a top level file so we'd need to include manually a parent

**Review by lucas-zimerman — APPROVED** [review]

File: src/Sentry/Sentry.csproj [hunk 1]
```diff
@@ -8,6 +8,16 @@
     <Description>Official SDK for Sentry - Open-source error tracking that helps developers monitor and fix crashes in real time.</Description>
     <NoWarn Condition="$(TargetFramework) == 'netstandard2.0'">$(NoWarn);RS0017</NoWarn>
     <DefineConstants>$(AdditionalConstants)</DefineConstants>
+    <Configurations>Debug;Release;Release_Unity;Debug_Unity</Configurations>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(Configuration) == 'Release_Unity'">
```
Comment (bruno-garcia) [inline src/Sentry/Sentry.csproj:14]: I believe it's best we add an additional Property instead so we can continue to use Configurion=Release
The reasoning is that there (?) properties and behavior tied to Configuration `Release` and we might be missing those due to introducing a new configuration. 

Can we just pass an additional properly: `dotnet msbuild /p:Unity=true` ?

Comment (SimonCropp) [reply src/Sentry/Sentry.csproj:14]: but that wont work when building from and ide

**Pull request closed without merging**